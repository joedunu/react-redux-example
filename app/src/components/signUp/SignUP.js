import React, { Component } from 'react'
import PropTypes from 'prop-types'

import UserForm from '../userForm/UserForm'

class SignUP extends Component {
  render () {
    return (
      <UserForm
        onSubmit={this.props.createUserRequest}
        title='Sign Up'
        goTo={this.props.goTo}
      />
    )
  }
}

SignUP.propTypes = {
  createUserRequest: PropTypes.func,
  goTo: PropTypes.func.isRequired
}

export default SignUP
