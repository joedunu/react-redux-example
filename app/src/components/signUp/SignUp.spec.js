import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'

import SignUP from './SignUP'

describe('User Registration Form', () => {
  const mockStore = configureMockStore([])
  let store
  let combinedReducersState = {
    form: {}
  }

  beforeEach(() => {
    store = mockStore(combinedReducersState)
  })

  test('should pass snapshot test', () => {
    const component = shallow(<SignUP
      store={store}
      goTo={jest.fn()}
    />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
