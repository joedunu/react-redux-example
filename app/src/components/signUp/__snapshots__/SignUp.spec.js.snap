// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Registration Form should pass snapshot test 1`] = `
<ReduxForm
  goTo={[Function]}
  title="Sign Up"
/>
`;
